### PR TITLE
fix(ci): free disk space

### PR DESCRIPTION
**Pull request by motongxue**
### Description

Fixes # [(bug(ci): ci failing due to incomplete docker image backup)](https://github.com/apache/apisix/issues/11068)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by shreemaan-abhishek** [comment]
The github action you added won't work: https://github.com/apache/apisix/actions/runs/8389436022

**Comment by motongxue** [comment]
@shreemaan-abhishek need an approval for workflows.

**Comment by shreemaan-abhishek** [comment]
@motongxue it looks like a flaky failure. I have re-ran that workflow.

Could you please take care of the following as well?
image

**Comment by motongxue** [comment]
> > @motongxue it looks like a flaky failure. I have re-ran that workflow.
> > Could you please take care of the following as well? image
> 
> If the GHE runner's disk is not enough, the step `save docker images` will always fail(no space left on device). So the cache in CI will make no sence. So I think free disk space is a better approach.

I think so.

**Review by starsz — APPROVED** [review]

File: ci/free_disk_space.sh [hunk 1]
```diff
@@ -25,7 +25,24 @@ echo "==========================================================================
 echo "Freeing up disk space on CI system"
 echo "=============================================================================="
 
+echo "Initial disk usage:"
 df -h
-echo "Removing unnecessary large directories"
-sudo rm -rf /usr/local/lib/android /usr/share/dotnet /usr/share/swift
+
+echo "Removing large directories and runtimes..."
+sudo rm -rf /usr/local/lib/android /usr/share/dotnet /opt/ghc /usr/local/.ghcup
```
Comment (shreemaan-abhishek) [inline ci/free_disk_space.sh:32]: why did you remove `/usr/share/swift`?

**Review by shreemaan-abhishek — APPROVED** [review]

File: ci/free_disk_space.sh (hunk 1 above)
Comment (shreemaan-abhishek) [inline ci/free_disk_space.sh:32]: why did you remove /usr/share/swift?

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

File: ci/free_disk_space.sh (hunk 1 above)
Comment (motongxue) [reply ci/free_disk_space.sh]: I have added it. please review again.

**Review by shreemaan-abhishek — APPROVED** [review]
